src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
So, the same - try to give more understandable names
Here you can rewrite this like that
But Why?
It's more readable now
So, we transferred the ternary operator into the callback
But, we can create a function helper for this & this code will be great)

```suggestion
  const filteredByFemaleGender = people.filter((person) => withChildren
    ? people.some((child) => child.mother === person.name)
    : person.sex === 'f'
  );
```